oap-server/server-configuration/configuration-etcd/src/test/java/org/apache/skywalking/oap/server/configuration/etcd/EtcdConfigWatcherRegisterTest.java

```diff
@@ -0,0 +1,131 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.configuration.etcd;
+
+import com.google.common.collect.Sets;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import mousio.client.promises.ResponsePromise;
+import mousio.etcd4j.EtcdClient;
+import mousio.etcd4j.promises.EtcdResponsePromise;
+import mousio.etcd4j.requests.EtcdKeyGetRequest;
+import mousio.etcd4j.responses.EtcdKeysResponse;
+import org.apache.skywalking.oap.server.configuration.api.ConfigTable;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.spy;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+
+/**
+ * @author Alan Lau
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({EtcdKeysResponse.class, EtcdUtils.class, EtcdClient.class, URI.class})
+@PowerMockIgnore({"javax.management.*"})
+public class EtcdConfigWatcherRegisterTest {
+
+    @Before
+    @Test
+    public void shouldReadConfigs() throws Exception {
+        final String group = "skywalking";
+        final String testKey1 = "receiver-trace.default.slowDBAccessThreshold";
+        final String testVal1 = "test";
+        final String testKey2 = "testKey";
+        final String testVal2 = "testVal";
+
+        final EtcdServerSettings mockSettings = mock(EtcdServerSettings.class);
+        when(mockSettings.getGroup()).thenReturn(group);
+        mockStatic(EtcdUtils.class);
+
+        List<URI> uris = mock(List.class);
+        when(EtcdUtils.parse(any())).thenReturn(uris);
+
+        final EtcdClient client = PowerMockito.mock(EtcdClient.class);
+        whenNew(EtcdClient.class).withAnyArguments().thenReturn(client);
+
+        URI uri = new URI("http://localhost:2379");
```

Comment:
And you should get the port from system properties, which is dynamically set by `docker-maven-plugin`, as for the host, it doesn't matter

Reply:
ok